### PR TITLE
CI examples

### PR DESCRIPTION
**Pull request by jpadams**
first three checkboxes of #3918

tested these manually. automated testing to come.

### CODE REVIEW

**Comment by jpadams** [comment]
@vikram-dagger already gave 👍 to this approach in case someone can approve it sooner.

File: docs/current/sdk/go/275922-guides.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ slug: /sdk/go/275922/guides
 
 # Guides
 
+- [Use Dagger Go SDK in CI](./guides/768421-go-ci.md)
```
Comment (vikram-dagger) [inline docs/current/sdk/go/275922-guides.md:7]:
```suggestion
- [Use the Dagger Go SDK in CI](./guides/768421-go-ci.md)
```

File: docs/current/sdk/go/guides/768421-go-ci.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+slug: /768421/go-ci
+displayed_sidebar: "current"
+---
+
+# Dagger Go SDK in CI
```
Comment (vikram-dagger) [inline docs/current/sdk/go/guides/768421-go-ci.md:6]:
```suggestion
# Use the Dagger Go SDK in CI
```

File: docs/current/sdk/nodejs/guides/114934-nodejs-ci.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+---
+slug: /114934/nodejs-ci
+displayed_sidebar: "current"
+---
+
+# Dagger Node.js SDK in CI
```
Comment (vikram-dagger) [inline docs/current/sdk/nodejs/guides/114934-nodejs-ci.md:6]:
```suggestion
# Use the Dagger Node.js SDK in CI
```

File: docs/current/sdk/nodejs/guides/114934-nodejs-ci.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+{label: 'GitLab', value: 'gitlab'},
+{label: 'Jenkins', value: 'jenkins'},
+]}>
+
+<TabItem value="github-actions">
+
+```yaml title=".github/workflows/dagger.yaml" file=../snippets/nodejs-ci/actions.yml
+```
+
+Ensure that your `package.json` contains `@dagger.io/dagger` which can be installed using the [documentation](https://docs.dagger.io/sdk/nodejs/835948/install).
```
Comment (vikram-dagger) [inline docs/current/sdk/nodejs/guides/114934-nodejs-ci.md:24]:
```suggestion
Ensure that your `package.json` contains `@dagger.io/dagger`, as described in the [SDK installation documentation](https://docs.dagger.io/sdk/nodejs/835948/install).
```

More descriptive link for SEO

Also could you please update the link to be a relative link?

Same comments as above for other tabs on this page.

File: docs/current/sdk/go/guides/768421-go-ci.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+</TabItem>
+
+<TabItem value="jenkins">
+
+```groovy title="Jenkinsfile" file=../snippets/go-ci/Jenkinsfile
+```
+
+Requires `docker` client and `go` installed on your Jenkins agent, a Docker host available (can be `docker:dind`), and agents labeled in Jenkins with `dagger`.
```
Comment (vikram-dagger) [inline docs/current/sdk/go/guides/768421-go-ci.md:45]: Would it be better to place this and other notes above the code block? My concern is that a user will copy-paste the code block without realizing there are additional requirements when this is placed below.

File: docs/current/sdk/nodejs/guides/114934-nodejs-ci.md (hunk 4 above)
Comment (vikram-dagger) [inline docs/current/sdk/nodejs/guides/114934-nodejs-ci.md:24]: See prev q regarding placement of requirements.

File: docs/current/sdk/nodejs/snippets/nodejs-ci/Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,14 @@
+pipeline {
+  agent { label 'dagger' }
+
+  stages {
+    stage("dagger") {
+      steps {
+        sh '''
+            npm ci
+            node index.mjs
```
Comment (vikram-dagger) [inline docs/current/sdk/nodejs/snippets/nodejs-ci/Jenkinsfile:9]: For the other SDKs, we use `main.*` as the script name, should we follow that pattern here and call this `main.mjs`?

File: docs/current/sdk/python/234291-guides.md [hunk 7]
```diff
@@ -4,5 +4,6 @@ slug: /sdk/python/234291/guides
 
 # Guides
 
+- [Use Dagger Python SDK in CI](./guides/454108-python-ci.md)
```
Comment (vikram-dagger) [inline docs/current/sdk/python/234291-guides.md:7]:
```suggestion
- [Use the Dagger Python SDK in CI](./guides/454108-python-ci.md)
```

File: docs/current/sdk/python/guides/454108-python-ci.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
+---
+slug: /454108/python-ci
+displayed_sidebar: "current"
+---
+
+# Dagger Python SDK in CI
```
Comment (vikram-dagger) [inline docs/current/sdk/python/guides/454108-python-ci.md:6]:
```suggestion
# Use the Dagger Python SDK in CI
```

File: docs/current/sdk/python/guides/454108-python-ci.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+</TabItem>
+
+<TabItem value="jenkins">
+
+```groovy title="Jenkinsfile" file=../snippets/python-ci/Jenkinsfile
+```
+
+Requires `docker` client and `python` installed on your Jenkins agent, a Docker host available (can be `docker:dind`), and agents labeled in Jenkins with `dagger`.
```
Comment (vikram-dagger) [inline docs/current/sdk/python/guides/454108-python-ci.md:45]: See previous q regarding placement of this note.

File: docs/current/sdk/go/guides/768421-go-ci.md (hunk 5 above)
Comment (jpadams) [reply docs/current/sdk/go/guides/768421-go-ci.md:45]: Leaving in place as is for visual harmony, but can change later if needed.

File: docs/current/sdk/nodejs/snippets/nodejs-ci/Jenkinsfile (hunk 6 above)
Comment (jpadams) [reply docs/current/sdk/nodejs/snippets/nodejs-ci/Jenkinsfile:9]: good idea. Will change in hero on webpage as well.

File: docs/current/sdk/python/guides/454108-python-ci.md (hunk 9 above)
Comment (jpadams) [reply docs/current/sdk/python/guides/454108-python-ci.md:45]: see above

File: docs/current/sdk/nodejs/guides/114934-nodejs-ci.md (hunk 4 above)
Comment (jpadams) [reply docs/current/sdk/nodejs/guides/114934-nodejs-ci.md]: see above

**Comment by jpadams** [comment]
Also, I used the longer descriptions of the CI pages in the guides "index" pages, but used shorter version in the CI guide itself because at 100% zoom on laptop, the text wrapping was not pretty.

File: docs/current/sdk/nodejs/snippets/nodejs-ci/Jenkinsfile (hunk 6 above)
Comment (slumbering) [reply docs/current/sdk/nodejs/snippets/nodejs-ci/Jenkinsfile]: nit: Like I said in the PR, using index.mjs is actually the common naming. main.mjs feels uncommon IMO

**Pull request merged**